<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>The Extended Integer (XInt) Library: The \c on_exception Function And \c -fno-exceptions Support</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="index.html">The Extended Integer (XInt) Library</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>The <code>on_exception</code> Function And <code>-fno-exceptions</code> Support </h1>  </div>
</div>
<div class="contents">
<p>The <code>on_exception</code> Function</p>
<p>The library can be set to call a function whenever it encounters an error that would normally throw an exception, by passing it to the <a class="el" href="namespaceboost_1_1xint.html#ac2b936b22084368c66e62f8181c08241">on_exception</a> function. The handler function is given the file name and line number where the error occurred, and the exception object that would normally be thrown. This is meant for the <code>-fno-exceptions</code> support (see below), but can also be used to report the error or throw a different exception. If the handler returns, then XInt will deal with the error by throwing the exception (or calling <code>abort()</code> if the <code>-fno-exceptions</code> support is used).</p>
<p><code>-fno-exceptions</code> Support</p>
<p>The library includes support for GCC's -fno-exceptions option. When compiled with that option, a program cannot use <code>try</code>, <code>throw</code>, or <code>catch</code> statements. On any error that would otherwise throw an exception, the library will call the user-supplied exception-handler function as normal (see above), and calling <code>abort()</code> if it returns.</p>
<p>It is possible to use <code>setjmp()/longjmp</code>() to recover from errors when using this feature, but note that doing so will often result in memory leaks because no destructors or other cleanup code is allowed to run.</p>
<p>To enable this feature, define the preprocessor macro <code>BOOST_XINT_NO_EXCEPTIONS</code> before including any XInt library header. </p>
</div>
<hr>
<p><i>&copy; Copyright Chad Nelson, 2010-2011. Distributed under the Boost Software
License, Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
<a href=http://www.boost.org/LICENSE_1_0.txt>http://www.boost.org/LICENSE_1_0.txt</a>)
</i></p>
<p><i>Documentation generated on Wed Mar 9 2011 22:56:48 by&nbsp;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a>
1.7.1</small></address></i></p>
